### PR TITLE
MGMT-18832: in case istallation stopped we should exit reconcile loop right away

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: controllers/conditions.go [hunk 1]
```diff
@@ -106,14 +106,14 @@ func (r *ImageClusterInstallReconciler) setImageReadyCondition(ctx context.Conte
 
 func (r *ImageClusterInstallReconciler) setHostConfiguredCondition(ctx context.Context, ici *v1alpha1.ImageClusterInstall, err error) error {
 	cond := hivev1.ClusterInstallCondition{
-		Type:    hivev1.ClusterInstallStopped,
-		Status:  corev1.ConditionFalse,
+		Type:    hivev1.ClusterInstallRequirementsMet,
```
Comment (carbonin) [inline controllers/conditions.go:109]: Why change this?

I would understand this change as a part of making BMH required, but does it have anything to do with exiting the loop early?

Comment (tsorya) [reply controllers/conditions.go:109]: As we talked, missing bmh should not stop installation but rather set the requirements are not yet met

File: controllers/imageclusterinstall_controller_test.go [hunk 2]
```diff
@@ -908,7 +909,6 @@ var _ = Describe("Reconcile", func() {
 
 		res, err := r.Reconcile(ctx, req)
 		Expect(err).NotTo(HaveOccurred())
-		Expect(res).To(Equal(ctrl.Result{}))
```
[the remark this answers is not in the document]
Comment (tsorya) [reply controllers/imageclusterinstall_controller_test.go:911]: So i believe we can push it here as is and i will change test in anycase in the next one

**Comment by carbonin** [comment]
Can you squash out the second commit or make the message more meaningful?

File: controllers/imageclusterinstall_controller_test.go [hunk 3]
```diff
@@ -1236,8 +1247,7 @@ var _ = Describe("Reconcile", func() {
 		Expect(c.Get(ctx, key, clusterInstall)).To(Succeed())
 
 		cond := findCondition(clusterInstall.Status.Conditions, hivev1.ClusterInstallCompleted)
-		Expect(cond).NotTo(BeNil())
-		Expect(cond.Reason).ToNot(Equal(v1alpha1.InstallTimedoutReason))
+		Expect(cond).To(BeNil())
```
[the remark this answers is not in the document]
Comment (tsorya) [reply controllers/imageclusterinstall_controller_test.go:1250]: actually it still does it, we won't set timeout if cluster was already installed

**Review by carbonin — APPROVED** [review]

**Pull request merged**